LSPS3/README.md

```diff
@@ -0,0 +1,655 @@
+imposed on the LSP) has passed.
+
+LSPs that implement this API SHOULD include mitigations to allow them to
+determine if clients can be trusted, or to detect and protect against attacks
+on the LSP.
+How those mitigations are performed is beyond the scope of this specification.
+
+### Amounts
+
+All amounts in this protocol are specified in millisatoshi.
+
+All millisatoshi values MUST be represented as a string and MUST NOT be
+represented as integer values.
+
+> **Rationale** JSON numbers are specified to only fit IEEE double-precision
+> numbers, which have a 53 bit mantissa.
+> The maximum number of millisatoshis (21 million BTC) cannot be specified in
+> 53 bits.
+> In addition, actual JSON parsers may use a smaller size for parsed integers;
+> for instance, as signed 32 bit numbers.
+> The largest safe value is 2,147,483,647; 2,147,483,647msat = BTC0.021,474,836,47
+> which is too low.
+
+### Actors
+
+'LSP' is the API provider.
+'Client' is the API client.
+'Payer' is who pays for the initial receive of the client.
+
+## Flow (LSP Trusts Client Model)
+
+Overview:
+
+* The client determines that it would like to receive over Lightning, but
+  it has no inbound capacity, and the client cannot pre-buy this capacity
+  (for instance, it also has no outbound capacity or onchain funds).
+* The client determines the parameters of a particular LSP via a `GET`
+  request.
+  * The LSP also indicates how long those parameters are valid for,
+    which also defines a timeout for the rest of this flow; if the
+    flow does not complete in that timeout, the parameters are no
+    longer valid and the LSP will refuse to open the channel.
+* The client requests for a JIT channel, specifying how much it will
+  receive, via a `POST` request.
+* The LSP returns an SCID, which identifies this request to the LSP.
+* The client generates an invoice, which includes the above SCID and the
+  LSP node ID as a routehint.
+* The client hands the invoice to whoever it will receive funds from.
+* The payment is forwarded to the LSP.
+* The LSP recognizes the next hop SCID as being a JIT channel request,
+  and opens a 0-confirmation channel to the client, which must be
+  connected to the LSP at that time.
+* The LSP forwards the payment to the client, deducting the channel
+  opening fee.
+* The client claims the payment.
+
+### 1. API Information
+
+`GET /lsp/jitchannel` is the entrypoint for each client using the API.
+It indicates supported versions of this protocol, as well as any limits the
+LSP imposes, and parameters for payment.
+
+The client MUST pull `GET /lsp/jitchannel` to read:
+
+- the supported `versions` of the LSP.
+- the `opening_fee` of the LSP and its related parameters.
+
+The client MUST then determine protocol compatibility, and also determine
+if the fee imposed by the LSP is acceptable.
+
+Example `GET /lsp/jitchannel` response:
+
+```JSON
+{
+    "versions": [1],
+    "opening_fee_params": {
```

Comment:
Instead of a single `opening_fee_params` object, provide an *array* of such objects in `opening_fee_params_menu`.  "LSP MUST provide at least one object in `opening_fee_params_menu`".  Client selects any one of these objects.